add predicate to GFF3Codec to give a chance to filter out some unused attributes

lindenb opened the pull request:
### Description

When using a GFF3Reader with DecodeDepth==DEEP, it may use a large amount of memory with attributes that will never be used ("version" ,"tag", etc...). This PR gives the GFF3Codec a chance to set a `Predicate<String>` to only keep a defined set of attributes.



the private attribute of ID_ATTRIBUTE_KEY and NAME_ATTRIBUTE_KEY Gff3BaseData was removed to check if the predicate does not remove them.

a new method `setFilterOutAttribute` was added to GFF3Codec

the `static` attribute of GFF3Codec.parseLine was removed

I added a test codecFilterOutFieldsTest


### Things to think about before submitting:
- [X] Make sure your changes compile and new tests pass locally.
- [X] Add new tests or update existing ones:
  - A bug fix should include a test that previously would have failed and passes now.
  - New features should come with new tests that exercise and validate the new functionality.
- [X] Extended the README / documentation, if necessary
- [X] Check your code style.
- [X] Write a clear commit title and message
  - The commit message should describe what changed and is targeted at htsjdk developers
  - Breaking changes should be mentioned in the commit message.

lbergelson reviewed (changes requested):
@lindenb Thank you, this seems like a good idea.  I have a few nitpicky comments, but look good overall.

lindenb commented:
@lbergelson thank you for your review. I moved the `static final String` from Gff3BaseData to Gff3Constants, and replaced `setFilterOutAttribute`  by a new constructor.
